### ISSUE TITLE
Error: encountered unexpected expression during compilation to Marabou

### ISSUE THREAD

**Issue by MatthewDaggitt** [vehicle-lang/vehicle#77]
```vcl
network f : Tensor Rat [2] -> Rat

spec : Prop
spec = forall x . f [x , 0.0] >= 0
```
crashes with the error `Error: encountered unexpected expression '(or ((!= x1) 0.0)) ((< y0) 0)' during compilation to Marabou`.

The problem is that when we insert the magic input variables, the left hand side of the implication equating the input variables no longer normalises to `true` and therefore after normalisation we get an additional `or` that we assume we'd already normalised out at this point.

**Comment by MatthewDaggitt** [comment]
Misdiagnosed this. The reason the example above is failing is because of a bug in the insertion of magic variables which is fixed in [1ffe727](https://github.com/vehicle-lang/vehicle/commit/1ffe727ee6caad4bece5225d091762375a3636c4). The following is a new example that exhibits this problem:
```
network encode : Tensor Rat [5] -> Tensor Rat [2]
network decode : Tensor Rat [2] -> Tensor Rat [5]

identity : Prop
identity = forall x . decode (encode x) == x
```

**Comment by MatthewDaggitt** [comment]
This issue appears to have resolved itself at some point.
